Check chart changes before running kind

scottrigby opened the pull request:
Use case and context: https://github.com/helm/kind-action/pull/6

scottrigby committed:
Initial test to check changes before running kind

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Test chart change

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Test chart change

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Update GitHub syntax

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Test chart change"

This reverts commit c300824594268c66cfc6a76c23167dc52e5a0407.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Revert "Test chart change""

This reverts commit 256909ba5fb06f8e674bae20e271d3738d12cdc4.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Syntax fix

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Test assumptions

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Fix faulty logic

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Revert "Revert "Test chart change"""

This reverts commit 6e74b4ccb65964d5ecd16f49124e328de0c48cbc.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Shorten a bit

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Revert "Revert "Revert "Test chart change""""

This reverts commit cd986b55988dcb99456151f70a7f43f291ab730e.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Bump version

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Revert "Revert "Revert "Revert "Test chart change"""""

This reverts commit d32872d532eef2d2a8a0cf4920e7d7611a1f916d.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Bump version"

This reverts commit 3f2e6b7b9381b893346e60a9b76eda0df513e16c.

Signed-off-by: Scott Rigby <scott@r6by.com>

unguiculus commented on .github/workflows/lint-test.yaml:
```diff
@@ -9,10 +9,20 @@ jobs:
+        id: check_charts
+        run: |
+          changes=$(git diff --find-renames --name-only "$(git rev-parse --abbrev-ref HEAD)" remotes/origin/master -- charts)
+          [ -n "$changes" ] && echo "::set-output name=changed::true" || echo "No chart changes"
```
I think that special kind of log format only works if you log with their toolkit from a JS action?

unguiculus replied:
Seems it does work, but you'd have to skip all subsequent steps.

scottrigby replied:
Yeah their documentation wasn't 100% clear on that to me either, but I found a fun issue about it with tests I was able to emulate: https://github.com/actions/toolkit/issues/96. I have tested it, and it only skips the correct step, not all subsequent ones. I left the multiple "revert" chart change commits in this PR just to show how it didn't work at points then worked with this specific syntax 😄

scottrigby replied:
Oh, did you mean we'd also have to add a similar check to the subsequent steps as well?
> Seems it does work, but you'd have to skip all subsequent steps.

If so, I didn't add that because ct `lint` and `install` commands already check this

unguiculus committed:
Update .gitignore

Signed-off-by: Reinhard Naegele <unguiculus@gmail.com>

scottrigby committed:
Dependent chart test (#7)

* Dependent chart test

Signed-off-by: Scott Rigby <scott@r6by.com>

* add ct config for dependent chart repo

Signed-off-by: Scott Rigby <scott@r6by.com>

* Add dependency chart repos. Requires helm client

Signed-off-by: Scott Rigby <scott@r6by.com>

* Increase ct helm install timeout. Previously redis master PVC did not bind in time

Signed-off-by: Scott Rigby <scott@r6by.com>

* Use most recent Helm 2 for chart-releaser

Signed-off-by: Scott Rigby <scott@r6by.com>

* Test helm/kind-action local path provisioner fix in master before next release

Signed-off-by: Scott Rigby <scott@r6by.com>

* Use new helm/kind-action release

Signed-off-by: Scott Rigby <scott@r6by.com>

* Update to new helm/chart-testing-action release

Signed-off-by: Scott Rigby <scott@r6by.com>

* Update helm/chart-releaser-action version

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Test PR to add kind GH actions helper output to helm/chart-testing-action

Signed-off-by: Scott Rigby <scott@r6by.com>

Test chart change

Signed-off-by: Scott Rigby <scott@r6by.com>

Test changes for PR update

Signed-off-by: Scott Rigby <scott@r6by.com>

ID typo

Signed-off-by: Scott Rigby <scott@r6by.com>

Revert "Test chart change"

This reverts commit d574a6fb4e758c46f1ed0a96cae446825e939edd.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Merge remote-tracking branch 'origin/master' into check_changes

scottrigby committed:
Test version bump after chart change

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby committed:
Revert "Test chart change"

This reverts commit 1594fd71975ef7146a8c8349f69cb698f558e02b.

Revert "Test version bump after chart change"

This reverts commit 34c4c6e07d976249e34c9271d2a67695ad86cbcb.

Signed-off-by: Scott Rigby <scott@r6by.com>

scottrigby commented:
OK tested this and it works 😺 I think this is much simpler and requires a minimal change to helm/charts-testing-action. Opened a PR https://github.com/helm/chart-testing-action/pull/15 once merged, we can create a new release for helm/chart-testing-action and I'll change to the correct repo and version in this PR

davidkarlsen commented on .github/workflows/lint-test.yaml:
```diff
@@ -9,15 +9,17 @@ jobs:
 
+      - name: Run chart-testing (lint)
+        id: lint
+        uses: scottrigby/chart-testing-action@help-kind-run-only-if-changed
```
seems to do the trick (https://github.com/helm/charts-repo-actions-demo/commit/ba9a59f761584bd09602be584e56f8ad9ca14cb7/checks?check_suite_id=388497336) - so maybe just update this PR with the official helm action and then we merge the two successively?

scottrigby replied:
Just merged and created next pre-release tag https://github.com/helm/chart-testing-action/releases/tag/v1.0.0-alpha.3 and updated to official helm action release in 4dcf446 .

scottrigby committed:
Use new helm/chart-testing-action release with merged fix

Signed-off-by: Scott Rigby <scott@r6by.com>
